src/components/visualizer/threads.tsx

```diff
@@ -18,18 +10,21 @@ export default function Threads({ data }: ThreadsProps): React.ReactElement {
 
   useEffect(() => {
     const trs = data.map(thread => {
-      const className = 'thread-row' + (thread.inProcessor ? ' current' : '');
+      const inProcessor = thread.state === 'running';
+      const active = thread.state === 'running' || thread.state === 'ready';
+      const complete = thread.state === 'completed';
+      const className = 'thread-row' + (inProcessor ? ' current' : '');
       return (
         <tr className={className} key={`Thread: ${thread.name}`} title={`Thread: ${thread.name}`}>
           <td className="thread-name">{thread.name}</td>
-          <td className="thread-active">{getDisplayValueForBool(thread.active)}</td>
-          <td className="thread-completed">{getDisplayValueForBool(thread.complete)}</td>
+          <td className="thread-active">{getDisplayValueForBool(active)}</td>
+          <td className="thread-completed">{getDisplayValueForBool(complete)}</td>
         </tr>
       );
     });
 
     setTableValues(trs);
-  }, data);
+  }, [data]);
```

Comment:
The functionality of this change needs to be tested to see if the desired original functionality is still present.

Desired: on change of the array, or any elements in the array, reEvaluate the table